src/bootstrap/native.rs

```diff
@@ -91,6 +91,91 @@ pub fn prebuilt_llvm_config(
     Err(Meta { stamp, build_llvm_config, out_dir, root: root.into() })
 }
 
+// modified from `check_submodule` and `update_submodule` in bootstrap.py
+#[allow(unused_must_use)]
+pub(crate) fn update_llvm_submodule(builder: &Builder<'_>) {
+    let llvm_project = &Path::new("src").join("llvm-project");
+    let root = &builder.config.src;
+
+    // `distcheck` makes a copy of the source tree with no git history.
+    // It manages its own submodules, so skip updating altogether.
+    if !root.join(".git").exists() || !root.join(".gitmodules").exists()
+        || t!(root.join(llvm_project).join(".git").metadata()).len() == 0 {
```

Comment:
build.rust_info.is_git() looks like the right way to check most of these, not sure on the .gitmodules check...

Reply:
This was super helpful, thank you so much :heart: that let me figure out the issue is that I need *different* checks in `maybe_update` than in `update` - maybe_update should only update it if it already exists, update() should update it even for empty directories.